cmd/abapEnvironmentRunATCCheck.go

```diff
@@ -0,0 +1,354 @@
+		return req, fmt.Errorf("Triggering ATC run failed: %w", err)
+	}
+	defer req.Body.Close()
+	return req, err
+}
+
+func fetchXcsrfToken(requestType string, details connectionDetailsHTTP, body []byte, client piperhttp.Sender) (string, error) {
+
+	log.Entry().WithField("ABAP Endpoint: ", details.URL).Info("Fetching Xcrsf-Token")
+
+	header := make(map[string][]string)
+	header["X-Csrf-Token"] = []string{details.XCsrfToken}
+	header["Accept"] = []string{"application/vnd.sap.atc.run.v1+xml"}
+
+	req, err := client.SendRequest(requestType, details.URL, bytes.NewBuffer(body), header, nil)
+	if err != nil {
+		return "", fmt.Errorf("Fetching Xcsrf-Token failed: %w", err)
+	}
+	defer req.Body.Close()
+	token := req.Header.Get("X-Csrf-Token")
+	return token, err
+}
+
+func checkHost(config abapEnvironmentRunATCCheckOptions, details connectionDetailsHTTP) (connectionDetailsHTTP, error) {
+
+	var err error
+
+	if config.Host == "" {
+		cfconfig := cloudfoundry.ServiceKeyOptions{
+			CfAPIEndpoint:     config.CfAPIEndpoint,
+			CfOrg:             config.CfOrg,
+			CfSpace:           config.CfSpace,
+			Username:          config.Username,
+			Password:          config.Password,
+			CfServiceInstance: config.CfServiceInstance,
+			CfServiceKey:      config.CfServiceKeyName,
+		}
+		if cfconfig.CfServiceInstance == "" || cfconfig.CfOrg == "" || cfconfig.CfAPIEndpoint == "" || cfconfig.CfSpace == "" || cfconfig.CfServiceKey == "" {
+			return details, errors.New("Parameters missing. Please provide EITHER the Host of the ABAP server OR the Cloud Foundry ApiEndpoint, Organization, Space, Service Instance and a corresponding Service Key for the Communication Scenario SAP_COM_0510")
+		}
+		var abapServiceKey cloudfoundry.ServiceKey
+		abapServiceKey, err = cloudfoundry.ReadServiceKeyAbapEnvironment(cfconfig, true)
+		if err != nil {
+			return details, fmt.Errorf("Reading Service Key failed: %w", err)
+		}
+		details.User = abapServiceKey.Abap.Username
+		details.Password = abapServiceKey.Abap.Password
+		details.URL = abapServiceKey.URL
+		return details, err
+	}
+	details.User = config.Username
+	details.Password = config.Password
+	details.URL = config.Host
+	return details, err
+}
+
+func pollATCRun(details connectionDetailsHTTP, body []byte, client piperhttp.Sender) (string, error) {
+
+	log.Entry().WithField("ABAP endpoint", details.URL).Info("Polling ATC run status")
+
+	for {
+		resp, err := getHTTPResponseATCRun("GET", details, nil, client)
+		if err != nil {
+			return "", fmt.Errorf("Getting HTTP response failed: %w", err)
+		}
+		bodyText, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return "", fmt.Errorf("Reading response body failed: %w", err)
+		}
+
+		x := new(Run)
+		xml.Unmarshal(bodyText, &x)
+		log.Entry().WithField("StatusCode", resp.StatusCode).Info("Status: " + x.Status)
+
+		if x.Status == "Not Created" {
+			return "", err
+		}
+		if x.Status != "Running" && x.Status != "Not Yet Started" && x.Status == "Completed" {
```

Comment:
I think x.Status != "Running" && x.Status != "Not Yet Started" is not necessary

Reply:
True since status can only have one assigned value. Removed the mentioned part accordingly.